var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -683,16 +683,16 @@ def cmake_args(self):
                 # hence the test to see if the version starts with "flang".
                 targets.append('CppBackend')
 
-            if 'x86' in spec.architecture.target.lower():
+            if 'x86' in spec.target.family:
```

Comment:
Checking if a string is in a microarchitecture means looking for features, for instance:
```python
'avx' in spec.target
```
If you want to check that the family is a given one you need to use the `==` operator:
```python
spec.target.family == 'x86_64'
```
The checks in this PR will all return `False`

Reply:
I see, thanks. But I don't think I can just use `spec.target.family == 'x'`. The original version of this code was `if 'x' in spec.target`, so if `x` is `arm`, the condition would be true for things like `armv7`, for example. If I use `spec.target.family == 'arm'` the condition will become false for `armv7`. Since I don't have a full list of the possible families, I don't know if the original author initially meant to have some of the conditions evaluate to true for a group of families. The two options I see here are (1) find a list of ALL the possible families and update this sequence of conditions accordingly, or (2) convert `spec.target.family` into a string so I can continue using `if 'x' in str(spec.target.family)`. What do you suggest?

Reply:
Also if I look at `lib/spack/llnl/util/cpu/microarchitecture.py` line 98 the quality function cannot be used to compare against a string object (or maybe I'm missing something).

Reply:
> Also if I look at lib/spack/llnl/util/cpu/microarchitecture.py line 98 the equality function cannot be used to compare against a string object (or maybe I'm missing something).

The decorator:

https://github.com/spack/spack/blob/ccfe8aaf76f8fc294ea1bf0a9a1cd829c4166c5f/lib/spack/llnl/util/cpu/microarchitecture.py#L24-L37

Reply:
I'm looking at the `llvm` package. Will post some suggestions in a while.